.github/workflows/build-rust-cross-platform.yml

```diff
@@ -0,0 +1,62 @@
+name: Build Rust Cross Platform
+
+on:
+  workflow_call:
+
+jobs:
+  build_rust:
+    strategy:
+      fail-fast: false
+      matrix:
+        settings:
+          - os: macos-12
+            target: x86_64-apple-darwin
+            arch: x64
+            folder_name: macos-x64
```

Comment:
I think we should try and use `target` for the uploaded paths, `folder_name` will be specific to the different SDK bindings.